src/Ethereum/RLP.cpp

```diff
@@ -134,3 +140,127 @@ Data RLP::putint(uint64_t i) noexcept {
     };
     // clang-format on
 }
+
+Data RLP::decodeRawTx(const Data& data) {
+    auto decoded = decode(data).decoded;
+    if (decoded.size() < 9) {
+        return {};
+    }
+    json result = {
+        {"nonce", hexEncoded(decoded[0])},
+        {"gasPrice", hexEncoded(decoded[1])},
+        {"gas", hexEncoded(decoded[2])},
+        {"to", hexEncoded(decoded[3])},
+        {"amount", hexEncoded(decoded[4])},
+        {"input", hexEncoded(decoded[5])},
+        {"v", hexEncoded(decoded[6])},
+        {"r", hexEncoded(decoded[7])},
+        {"s", hexEncoded(decoded[8])},
+    };
+    auto dumped = result.dump();
+    return Data(dumped.begin(), dumped.end());
+}
+
+
+RLP::DecodedItem RLP::decode(const Data& input) {
+    if (input.size() == 0) {
+        throw std::invalid_argument("can't decode empty rlp data");
+    }
+    RLP::DecodedItem item;
+    auto inputLen = input.size();
+    auto prefix = input[0];
+    if (prefix <= 0x7f) {
+        // a single byte whose value is in the [0x00, 0x7f] range, that byte is its own RLP encoding.
+        item.decoded.push_back(Data{input[0]});
+        item.remainder = Data(input.begin() + 1, input.end());
+        return item;
+    } else if (prefix <= 0xb7) {
```

Comment:
All if branches end with a return; else is unnecessary.

Reply:
thanks, all done